lib/services/downloads_service_backend.dart

```diff
@@ -0,0 +1,1515 @@
+        missingChildIds.length);
+    assert(
+        missingChildIds.length + oldChildIds.length - childrenToUnlink.length ==
+            children.length);
+    _isar.downloadItems.putAllSync(childrenToPutAndLink);
+    _downloadsService.deleteBuffer
+        .addAll(childrenToUnlink.map((e) => e.isarId));
+    if (missingChildIds.isNotEmpty || childrenToUnlink.isNotEmpty) {
+      links.updateSync(
+          link: childrenToLink + childrenToPutAndLink,
+          unlink: childrenToUnlink);
+      // Collection download state may need changing with different children
+      _downloadsService.syncItemState(parent);
+    }
+    return (
+      childrenToPutAndLink.map((e) => e.isarId).toSet(),
+      childIdsToLink.toSet(),
+      childIdsToUnlink
+    );
+  }
+
+  /// Get BaseItemDto from the given collection ID.  Tries local cache, then
+  /// Isar, then requests data from jellyfin in a batch with other calls
+  /// to this method.  Used within [_syncDownload].
+  Future<DownloadStub?> _getCollectionInfo(
+      String id, DownloadItemType type, bool forceServer) async {
+    if (_metadataCache.containsKey(id)) {
+      return _metadataCache[id];
+    }
+    Completer<DownloadStub?> itemFetch = Completer();
+    try {
+      DownloadStub? item;
+      if (!forceServer) {
+        item = _isar.downloadItems.getSync(DownloadStub.getHash(id, type));
+        if (item != null) {
+          return item;
+        }
+      }
+      _metadataCache[id] = itemFetch.future;
+      item = await _jellyfinApiData
+          .getItemByIdBatched(
+              id, "${_jellyfinApiData.defaultFields},childCount,sortName")
+          .then((value) => value == null
+              ? null
+              : DownloadStub.fromItem(item: value, type: type));
+      _downloadsService.resetConnectionErrors();
+      itemFetch.complete(item);
+      return itemFetch.future;
+    } catch (e) {
+      // Retries should try connecting again instead of re-using error
+      _metadataCache.remove(id);
+      itemFetch.completeError(e);
+      _downloadsService.incrementConnectionErrors();
+      return itemFetch.future;
+    }
+  }
+
+  // These cache downloaded metadata during _syncDownload
+  Map<String, Future<DownloadStub?>> _metadataCache = {};
+  Map<String, Future<List<String>>> _childCache = {};
+
+  /// Get ordered child items for the given collection DownloadStub.  Tries local
+  /// cache, then requests data from jellyfin.  Used within [_syncDownload].
+  Future<List<DownloadStub>> _getCollectionChildren(DownloadStub parent) async {
+    DownloadItemType childType;
+    BaseItemDtoType childFilter;
+    String? fields;
+    assert(parent.type == DownloadItemType.collection);
+    switch (parent.baseItemType) {
+      case BaseItemDtoType.playlist || BaseItemDtoType.album:
+        childType = DownloadItemType.song;
+        childFilter = BaseItemDtoType.song;
+        fields = "${_jellyfinApiData.defaultFields},MediaSources,SortName";
```

Comment:
Does Jellyfin gracefully handle duplicate fields? Otherwise, building a list/set and then `join()`ing it might be better here

Reply:
It does.  I've had duplicates with the defaults in this list and everything still worked fine.